lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -48,9 +52,12 @@ defmodule RealtimeWeb.RealtimeChannel do
          :ok <- limit_joins(socket.assigns),
          :ok <- limit_channels(socket),
          :ok <- limit_max_users(socket.assigns),
-         {:ok, claims, confirm_token_ref} <- confirm_token(socket),
-         {:ok, _} <- Connect.lookup_or_start_connection(tenant),
-         is_new_api <- is_new_api(params) do
+         {:ok, claims, confirm_token_ref, access_token} <- confirm_token(socket),
+         {:ok, db_conn} <- Connect.lookup_or_start_connection(tenant),
+         is_new_api = is_new_api(params),
```

Comment:
I would move `is_new_api(params)` inside the `do`-block. It does not seem to be used by `<-` clauses.